Linking Error: Undefined Symbols Due to Missing AVFoundation Framework, Breaking iOS Build

running `make run` command to build for ios target fails due to linking error. probably related to  #128.


❌ Error Message:

~~~make
Undefined symbols for architecture arm64:
  "_AVAudioSessionCategoryAmbient", referenced from:
      _main_rs in libmobile.a[2](mobile-c637714a32ea4ff4.0s3a5vvu0qtad299j2alsro0q.rcgu.o)
ld: symbol(s) not found for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)

** BUILD FAILED **
~~~

✅  Fix:

Link the AVFoundation framework in Xcode:

1. Open the .xcodeproj file.
2. In the Project Navigator, select "mobile" project, under "Targets" select "mobile"
3. Go to the "Build Phases" tab and expand "Link Binary with Libraries."
4. Click the + button, search for "AVFoundation," and add AVFoundation.framework.
5. Ensure it's marked as "Required."
6. Clean the project (Shift + Command + K) and rebuild.

--- a/mobile/mobile.xcodeproj/project.pbxproj
+++ b/mobile/mobile.xcodeproj/project.pbxproj
@@ -17,6 +17,7 @@
 		57CD6306253C7A940098CD4A /* AudioToolbox.framework in Frameworks */ = {isa = PBXBuildFile; fileRef = 57CD6305253C7A940098CD4A /* AudioToolbox.framework */; };
 		57CD630E253C80EC0098CD4A /* assets in Resources */ = {isa = PBXBuildFile; fileRef = 57CD630A253C7F5F0098CD4A /* assets */; };
 		6ADF1AB92CCDA73A00AF5F8E /* QuartzCore.framework in Frameworks */ = {isa = PBXBuildFile; fileRef = 6ADF1AB82CCDA73A00AF5F8E /* QuartzCore.framework */; };
+		D4A53EFF2DDD2FF70035BC01 /* AVFoundation.framework in Frameworks */ = {isa = PBXBuildFile; fileRef = D4A53EFE2DDD2FF70035BC01 /* AVFoundation.framework */; };
 /* End PBXBuildFile section */
 
 /* Begin PBXContainerItemProxy section */
@@ -42,6 +43,7 @@
 		9F1B41978FA53999AA836D0F /* Security.framework */ = {isa = PBXFileReference; lastKnownFileType = wrapper.framework; name = Security.framework; path = System/Library/Frameworks/Security.framework; sourceTree = SDKROOT; };
 		A39528EB2CCB182F5328223A /* libc++.tbd */ = {isa = PBXFileReference; lastKnownFileType = "sourcecode.text-based-dylib-definition"; name = "libc++.tbd"; path = "usr/lib/libc++.tbd"; sourceTree = SDKROOT; };
 		AF7DE91055EBD05ED77E57F9 /* main.m */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.c.objc; path = main.m; sourceTree = "<group>"; };
+		D4A53EFE2DDD2FF70035BC01 /* AVFoundation.framework */ = {isa = PBXFileReference; lastKnownFileType = wrapper.framework; name = AVFoundation.framework; path = System/Library/Frameworks/AVFoundation.framework; sourceTree = SDKROOT; };
 		FE71FBCAA714DB4F42459106 /* UIKit.framework */ = {isa = PBXFileReference; lastKnownFileType = wrapper.framework; name = UIKit.framework; path = System/Library/Frameworks/UIKit.framework; sourceTree = SDKROOT; };
 /* End PBXFileReference section */
 
@@ -55,6 +57,7 @@
 				134866208A035F8615C99114 /* Metal.framework in Frameworks */,
 				2604C99FAB5A8322EDCABB9F /* UIKit.framework in Frameworks */,
 				55B7188F81C3C4183F81D3AE /* libc++.tbd in Frameworks */,
+				D4A53EFF2DDD2FF70035BC01 /* AVFoundation.framework in Frameworks */,
 				57CD6306253C7A940098CD4A /* AudioToolbox.framework in Frameworks */,
 			);
 			runOnlyForDeploymentPostprocessing = 0;
@@ -95,6 +98,7 @@
 		EB028409C2D0655412DA6E44 /* Frameworks */ = {
 			isa = PBXGroup;
 			children = (
+				D4A53EFE2DDD2FF70035BC01 /* AVFoundation.framework */,
 				6ADF1AB82CCDA73A00AF5F8E /* QuartzCore.framework */,
 				57CD6305253C7A940098CD4A /* AudioToolbox.framework */,
 				A39528EB2CCB182F5328223A /* libc++.tbd */,

--- a/mobile/src/lib.rs
+++ b/mobile/src/lib.rs
@@ -8,6 +8,8 @@ unsafe extern "C" fn main_rs() {
     main();
 }
 
+// this macro is a no-op on ios and only needed for anroid since bevy 0.16
+// see https://github.com/bevyengine/bevy/pull/14780
 #[bevy_main]
 fn main() {
     #[cfg(target_os = "ios")]

--- a/mobile/src/lib.rs
+++ b/mobile/src/lib.rs
@@ -8,6 +8,7 @@ unsafe extern "C" fn main_rs() {
     main();
 }
 
+#[bevy_main]
 fn main() {
     #[cfg(target_os = "ios")]
     unsafe {

--- a/mobile/src/lib.rs
+++ b/mobile/src/lib.rs
@@ -3,7 +3,11 @@ use bevy::window::WindowMode;
 use bevy::winit::WinitSettings;
 use bevy_game::GamePlugin; // ToDo: Replace bevy_game with your new crate name.
 
-#[bevy_main]
+#[unsafe(no_mangle)]
+unsafe extern "C" fn main_rs() {
+    main();
+}
+
 fn main() {
     #[cfg(target_os = "ios")]
     unsafe {

